documentation/bolt_defaults_reference.md.erb

```diff
@@ -0,0 +1,92 @@
+# `bolt-defaults.yaml` options
```

Comment:
Based on feedback in the community Slack, a list of options seems to be preferred over a table of options. Users find it easier to find what they need and like being able to link to individual options since the headers automatically receive anchors.